metricbeat/module/windows/perfmon/pdh_windows.go

```diff
@@ -80,24 +85,43 @@ func PdhGetFormattedCounterValue(counter PdhCounterHandle, format PdhCounterForm
 	return counterType, &value, nil
 }
 
-func PdhGetFormattedCounterArray(counter PdhCounterHandle, format PdhCounterFormat) (int, *[]PdhCounterValueItem, error) {
+func PdhGetFormattedCounterArray(counter PdhCounterHandle, format PdhCounterFormat) ([]CounterValueItem, error) {
 	var bufferSize uint32
 	var bufferCount uint32
 
 	if err := _PdhGetFormattedCounterArray(counter, format, &bufferSize, &bufferCount, nil); err != nil {
 		//From MSDN: You should call this function twice, the first time to get the required buffer size (set ItemBuffer to NULL and lpdwBufferSize to 0), and the second time to get the data.
 		if PdhErrno(err.(syscall.Errno)) == PDH_MORE_DATA {
-			value := make([]PdhCounterValueItem, bufferSize)
-			if err := _PdhGetFormattedCounterArray(counter, format, &bufferSize, &bufferCount, &value[0]); err != nil {
-				return 0, nil, PdhErrno(err.(syscall.Errno))
+			pdhValues := make([]PdhCounterValueItem, bufferSize)
+			if err := _PdhGetFormattedCounterArray(counter, format, &bufferSize, &bufferCount, &pdhValues[0]); err != nil {
+				return nil, PdhErrno(err.(syscall.Errno))
+			}
+
+			values := make([]CounterValueItem, bufferCount)
+			var i uint32
+
+			for i = 0; i < bufferCount; i++ {
+				value := CounterValueItem{}
+				a := (*[1<<30 - 1]uint16)(unsafe.Pointer(pdhValues[i].SzName))
```

Comment:
We should try to find a more "safe" way to do this in the future. Using the unsafe pointer to cast an arbitrary size array worries me. But that's for another PR another time. :)